lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -32,20 +32,17 @@ static void RecyclerExecuteICUWithRetry(_In_ ICUFunc func, _In_ Recycler *recycl
     AssertOrFailFastMsg(success, "Could not allocate buffer for ICU call");
 }
 
-#define ICU_ERROR_FMT _u("INTL: %S failed with error code %S\n")
-#define ICU_EXPR_FMT _u("INTL: %S failed expression check %S\n")
+#define INTL_TRACE(fmt, ...) Output::Trace(Js::IntlPhase, _u("%S(): " fmt "\n"), __func__, __VA_ARGS__)
```

Comment:
This might need to use `##__VA_ARGS_` instead of `__VA_ARGS__`

Reply:
Although reading further, `##__VA_ARGS__` might be a gcc non-standard extension. The reason that this is failing in the no-arguments case is that it's expanding to `..., __func__, )`, note the trailing comma. VC fixes that for you, but gcc/clang don't by default, unless you use the `##__VA_ARGS__` macro.

Reply:
Does `##__VA_ARGS_` work with Clang too?
I wonder if you could you perhaps do something like this? (though it is definitely less than ideal...)
```
#ifdef (_MSC_VER)
#define INTL_TRACE(fmt, ...) Output::Trace(Js::IntlPhase, _u("%S(): " fmt "\n"), __func__, __VA_ARGS__) 
#else
#define INTL_TRACE(fmt, ...) Output::Trace(Js::IntlPhase, _u("%S(): " fmt "\n"), __func__, ##__VA_ARGS__) 
#endif
```

Reply:
@jackhorton How was this resolved?

---
In reply to: [177937964](https://github.com/Microsoft/ChakraCore/pull/4894#discussion_r177937964) [](ancestors = 177937964)

Reply:
I just added an argument to the one callsite that didn't have arguments :) I am happy to punt on this until later.